### ISSUE TITLE
Use redux/something redux-like for better state management

### ISSUE THREAD

**Issue by dwilliams27** [ColemanMitch/Spowerfy-Redux#24]
https://react-redux.js.org/

Once our state gets too complicated, we'll need to move to some sort of state management system (with actions/reducers/effects/etc). I'm familiar with Redux so could take a stab at it; if you're not familiar with it it will definitely have a learning curve though

**Comment by ColemanMitch** [comment]
If we're going to implement Redux, what would your thoughts be on conveying our class components to functional components 👀

**Comment by dwilliams27** [comment]
Honestly I'm just SO much more used to class components, since there's no such thing as a functional component in Angular. I'd probably say it'd be more trouble than its worth switching over, but if you're super into the idea I'm not entirely opposed to it (especially for small components)

**Comment by dwilliams27** [comment]
Does redux work better with functional components or smth?

**Comment by ColemanMitch** [comment]
> Honestly I'm just SO much more used to class components, since there's no such thing as a functional component in Angular. I'd probably say it'd be more trouble than its worth switching over, but if you're super into the idea I'm not entirely opposed to it (especially for small components)

Yeah, that's fair. I'm more comfortable with functional components so I may just create a new branch to fiddle around with converting our components over

**Comment by ColemanMitch** [comment]
> Does redux work better with functional components or smth?

I can't really speak to that since I've like barely ever used Redux
